E3Next/Processors/Bots.cs

```diff
@@ -770,6 +772,44 @@ public void Trade(string name)
             string command = "/notify TradeWnd TRDW_Trade_Button leftmouseup";
 			PubServer.AddTopicMessage("OnCommand-" + name, $"{E3.CurrentName}:{false}:{command}");
 		}
+
+        public void BroadcastLootingCorpse(int corpseId)
+        {
+            PubServer.AddTopicMessage("${Me.Looting}", $"{corpseId}");
+        }
+
+        Dictionary<string, CircularBuffer<int>> _lootCollection = new Dictionary<string, CircularBuffer<int>>();
+        public bool HasGroupLooted(int corpseId)
+        {
+            string keyToUse = "${Me.Looting}";
+            var allTopics = NetMQServer.SharedDataClient.TopicUpdates;
+
+            foreach (var userTopics in allTopics.Where(x => x.Key != E3.CurrentName))
+            {
+                if (!userTopics.Value.ContainsKey(keyToUse))
+                {
+                    continue;
+                }
+                var entry = userTopics.Value[keyToUse];
+                if (!_lootCollection.ContainsKey(userTopics.Key))
+                {
+                    _lootCollection.Add(userTopics.Key, new CircularBuffer<int>(10));
+                }
+                var sharedInfo = _lootCollection[userTopics.Key];
+
```

Comment:
I think we should check here if the last update was modified? that way we don't keep pushing the same values in?

Reply:
the other thing we have to think about is our checking vs when the value has been updated (basically going to miss things), I believe the check is done. honestly I will download your PR and puzzle it out tomorrow to give suggestions on how to do this. (the code compare in github is less than desirable sometimes :) )

Reply:
Ah yea, we should just be able to check if it's already there and only add it if not?

Reply:
Its a bit more complicated than that, I created a branch of some code I just put together, it has ZERO testing and I only confirmed it compiles. 

https://github.com/RekkasGit/E3Next/tree/feature/group_loot_check

Basically its a lot like how buffs are handled, each toon will keep a list of corpse IDs

Toon1: "1,2,3,4,5,6"
Toon2: "7,8,9,10,11"

This list will be transferred to all the clients (on a separate thread) as a comma delimited string as shown above.  Serialization/Deserialization only happens when the list is updated and then published out to all other clients. 
On each toon these lists will be as a List<Int32> and only updated when needed based off the backend string when its updated.

It needs initial testing/debugging to make sure everything is correct which I will see about getting to possibly/maybe tomorrow (bday, so might not!)

Reply:
Happy birthday! No rush on it :). I looked at your approach and I think both approaches will yield basically the same results assuming all messages are sent/received.

Reply:
Ty! I slept in waaay too late.

The biggest challenge with the approach currently on is missing data. New data only appears when looting corpses and another party member could have looted 2 corpses by that time and the code will only see the last one. 

Also 

EventProcessor.ProcessEventsInQueues("${Me.Looting}");

Isn't doing what you think its doing! This is to process commands from EQ into E3 as when commands are issued they drop into a buffer that has to be processed. It does nothing on the Shared Data. 

So the version I am working on works like the buffs, you just get a snapshot of what the other bots have worked on always updated in memory, if your checking for it or not, as it is kept up to date via the ShareData thread, if EQ is processing or not.  Then on the main bot side, as we go to loot a corpse we check to see vs that up to date shared data of all data entries.

Reply:
Ah that makes sense, thanks for the explanation!